### DISCUSSION TITLE
How to prune useless rules?

### DISCUSSION THREAD

**Discussion by bpdot** [rasbt/mlxtend#793] [Q&A]
antecedents | consequents | antecedent support | consequent support | support | confidence | lift | leverage | conviction | antecedent_len
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
(banana) | (eggs) | 0.054217 | 0.427711 | 0.054217 | 1.0 | 2.338028 | 0.031028 | inf | 1
(banana, apple) | (eggs) | 0.054217 | 0.427711 | 0.054217 | 1.0 | 2.338028 | 0.031028 | inf | 2

The "(banana, apple) -> (eggs)" rule doesn't improve confidence compare to  "(banana) -> (eggs)" rule. 
So I want to prune  "(banana, apple) -> (eggs)" rule. 
How can I do that?

**Comment by rasbt** [comment]
There is no specific API for pruning. You can use the pandas API on the resulting data frame to remove individual rows. E.g.,

```python
selection = (df['antecedents'] == frozenset({'banana', 'apple'})) & (df['consequents'] == frozenset({'eggs'}))
new_df = df.loc[ ~selection ]
```

**Reply by bpdot** [reply]
Thanks for your reply. 
There are many useless rules to be pruned. So filtering them one by one may not be acceptable.

**Reply by rasbt** [reply]
Yeah, having some automated pruning might be a good enhancement

**Reply by rasbt** [reply]
Added it to the issue tracker at #795
